### ISSUE TITLE
'Enabling 'ingress' returned an error'

### ISSUE THREAD

**Issue by tlemons** [kubernetes/minikube#16896]
### What Happened?

Hi - I'm attempting to start minikube with the ingress addon and am seeing a time-out error:

ldpdd040:/tmp # sudo minikube delete
* Deleting "minikube" in docker ...
* Deleting container "minikube" ...
* Removing /root/.minikube/machines/minikube ...
* Removed all traces of the "minikube" cluster.

ldpdd040:/tmp # minikube start --cpus=4 --memory=6g  --force
* minikube v1.30.1 on Suse 12.5
! minikube skips various validations when --force is supplied; this may lead to unexpected behavior
* Automatically selected the docker driver. Other choices: none, ssh
* The "docker" driver should not be used with root privileges. If you wish to continue as root, use --force.
* If you are running minikube within a VM, consider using --driver=none:
*   https://minikube.sigs.k8s.io/docs/reference/drivers/none/
! docker is currently using the btrfs storage driver, consider switching to overlay2 for better performance
* Using Docker driver with root privileges
* Starting control plane node minikube in cluster minikube
* Pulling base image ...
* Creating docker container (CPUs=4, Memory=6144MB) ...
* Preparing Kubernetes v1.26.3 on Docker 23.0.2 ...
  - kubelet.localStorageCapacityIsolation=false
  - Generating certificates and keys ...
  - Booting up control plane ...
  - Configuring RBAC rules ...
* Configuring bridge CNI (Container Networking Interface) ...
  - Using image gcr.io/k8s-minikube/storage-provisioner:v5
* Verifying Kubernetes components...
* Enabled addons: storage-provisioner, default-storageclass
* kubectl not found. If you need it, try: 'minikube kubectl -- get pods -A'
* Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default

ldpdd040:/tmp # minikube addons enable ingress --alsologtostderr
I0717 07:41:17.754551 1135490 out.go:296] Setting OutFile to fd 1 ...
I0717 07:41:17.754914 1135490 out.go:343] TERM=xterm,COLORTERM=1, which probably does not support color
I0717 07:41:17.754934 1135490 out.go:309] Setting ErrFile to fd 2...
I0717 07:41:17.754943 1135490 out.go:343] TERM=xterm,COLORTERM=1, which probably does not support color
I0717 07:41:17.755085 1135490 root.go:336] Updating PATH: /root/.minikube/bin
I0717 07:41:17.759631 1135490 out.go:177] * ingress is an addon maintained by Kubernetes. For any concerns contact minikube on GitHub.
You can view the list of minikube maintainers at: https://github.com/kubernetes/minikube/blob/master/OWNERS
* ingress is an addon maintained by Kubernetes. For any concerns contact minikube on GitHub.
You can view the list of minikube maintainers at: https://github.com/kubernetes/minikube/blob/master/OWNERS
I0717 07:41:17.822955 1135490 config.go:182] Loaded profile config "minikube": Driver=docker, ContainerRuntime=docker, KubernetesVersion=v1.26.3
I0717 07:41:17.822998 1135490 addons.go:66] Setting ingress=true in profile "minikube"
I0717 07:41:17.823010 1135490 addons.go:228] Setting addon ingress=true in "minikube"
I0717 07:41:17.823050 1135490 host.go:66] Checking if "minikube" exists ...
I0717 07:41:17.823744 1135490 cli_runner.go:164] Run: docker container inspect minikube --format={{.State.Status}}
I0717 07:41:17.849117 1135490 out.go:177]   - Using image registry.k8s.io/ingress-nginx/kube-webhook-certgen:v20230312-helm-chart-4.5.2-28-g66a760794
  - Using image registry.k8s.io/ingress-nginx/kube-webhook-certgen:v20230312-helm-chart-4.5.2-28-g66a760794
I0717 07:41:17.852569 1135490 out.go:177]   - Using image registry.k8s.io/ingress-nginx/kube-webhook-certgen:v20230312-helm-chart-4.5.2-28-g66a760794
  - Using image registry.k8s.io/ingress-nginx/kube-webhook-certgen:v20230312-helm-chart-4.5.2-28-g66a760794
I0717 07:41:17.864524 1135490 out.go:177]   - Using image registry.k8s.io/ingress-nginx/controller:v1.7.0
  - Using image registry.k8s.io/ingress-nginx/controller:v1.7.0
I0717 07:41:17.868752 1135490 addons.go:420] installing /etc/kubernetes/addons/ingress-deploy.yaml
I0717 07:41:17.868794 1135490 ssh_runner.go:362] scp memory --> /etc/kubernetes/addons/ingress-deploy.yaml (16145 bytes)
I0717 07:41:17.868874 1135490 cli_runner.go:164] Run: docker container inspect -f "'{{(index (index .NetworkSettings.Ports "22/tcp") 0).HostPort}}'" minikube
I0717 07:41:17.893489 1135490 sshutil.go:53] new ssh client: &{IP:127.0.0.1 Port:32792 SSHKeyPath:/root/.minikube/machines/minikube/id_rsa Username:docker}
I0717 07:41:17.995072 1135490 ssh_runner.go:195] Run: sudo KUBECONFIG=/var/lib/minikube/kubeconfig /var/lib/minikube/binaries/v1.26.3/kubectl apply -f /etc/kubernetes/addons/ingress-deploy.yaml
I0717 07:41:18.797413 1135490 addons.go:464] Verifying addon ingress=true in "minikube"
I0717 07:41:18.819637 1135490 out.go:177] * Verifying ingress addon...
* Verifying ingress addon...
I0717 07:41:18.848395 1135490 kapi.go:75] Waiting for pod with label "app.kubernetes.io/name=ingress-nginx" in ns "ingress-nginx" ...
I0717 07:41:18.891482 1135490 kapi.go:86] Found 3 Pods for label selector app.kubernetes.io/name=ingress-nginx
I0717 07:41:18.891510 1135490 kapi.go:96] waiting for pod "app.kubernetes.io/name=ingress-nginx", current state: Pending: [<nil>]
I0717 07:41:19.398905 1135490 kapi.go:96] waiting for pod "app.kubernetes.io/name=ingress-nginx", current state: Pending: [<nil>]
I0717 07:41:19.894803 1135490 kapi.go:96] waiting for pod "app.kubernetes.io/name=ingress-nginx", current state: Pending: [<nil>]
I0717 07:41:20.395648 1135490 kapi.go:96] waiting for pod "app.kubernetes.io/name=ingress-nginx", current state: Pending: [<nil>]
I0717 07:41:21.065189 1135490 kapi.go:96] waiting for pod "app.kubernetes.io/name=ingress-nginx", current state: Pending: [<nil>]
I0717 07:41:21.394744 1135490 kapi.go:96] waiting for pod "app.kubernetes.io/name=ingress-nginx", current state: Pending: [<nil>]
I0717 07:41:21.895536 1135490 kapi.go:96] waiting for pod "app.kubernetes.io/name=ingress-nginx", current state: Pending: [<nil>]
I0717 07:41:22.395156 1135490 kapi.go:96] waiting for pod "app.kubernetes.io/name=ingress-nginx", current state: Pending: [<nil>]
I0717 07:41:22.894992 1135490 kapi.go:96] waiting for pod "app.kubernetes.io/name=ingress-nginx", current state: Pending: [<nil>]
.
.
.
I0717 07:47:18.895394 1135490 kapi.go:96] waiting for pod "app.kubernetes.io/name=ingress-nginx", current state: Pending: [<nil>]
I0717 07:47:18.898011 1135490 kapi.go:96] waiting for pod "app.kubernetes.io/name=ingress-nginx", current state: Pending: [<nil>]
I0717 07:47:18.898037 1135490 kapi.go:107] duration metric: took 6m0.049655352s to wait for app.kubernetes.io/name=ingress-nginx ...
I0717 07:47:18.942609 1135490 out.go:177]

W0717 07:47:18.948180 1135490 out.go:239] X Exiting due to MK_ADDON_ENABLE: enable failed: run callbacks: running callbacks: [waiting for app.kubernetes.io/name=ingress-nginx pods: timed out waiting for the condition]
X Exiting due to MK_ADDON_ENABLE: enable failed: run callbacks: running callbacks: [waiting for app.kubernetes.io/name=ingress-nginx pods: timed out waiting for the condition]
W0717 07:47:18.948203 1135490 out.go:239] *
*
W0717 07:47:18.949769 1135490 out.go:239] ╭─────────────────────────────────────────────────────────────────────────────────────────────╮
│                                                                                             │
│    * If the above advice does not help, please let us know:                                 │
│      https://github.com/kubernetes/minikube/issues/new/choose                               │
│                                                                                             │
│    * Please run `minikube logs --file=logs.txt` and attach logs.txt to the GitHub issue.    │
│    * Please also attach the following file to the GitHub issue:                             │
│    * - /tmp/minikube_addons_5e63cb8ceb2e7e06e33b8dd7a9dcbebc0ad5ec5a_0.log                  │
│                                                                                             │
╰─────────────────────────────────────────────────────────────────────────────────────────────╯
╭─────────────────────────────────────────────────────────────────────────────────────────────╮
│                                                                                             │
│    * If the above advice does not help, please let us know:                                 │
│      https://github.com/kubernetes/minikube/issues/new/choose                               │
│                                                                                             │
│    * Please run `minikube logs --file=logs.txt` and attach logs.txt to the GitHub issue.    │
│    * Please also attach the following file to the GitHub issue:                             │
│    * - /tmp/minikube_addons_5e63cb8ceb2e7e06e33b8dd7a9dcbebc0ad5ec5a_0.log                  │
│                                                                                             │
╰─────────────────────────────────────────────────────────────────────────────────────────────╯
I0717 07:47:18.982579 1135490 out.go:177]

ldpdd040:/tmp #

Please let me know what other useful information I can provide.  Thanks!

### Attach the log file

[minikube_addons_5e63cb8ceb2e7e06e33b8dd7a9dcbebc0ad5ec5a_0.log](https://github.com/kubernetes/minikube/files/12068866/minikube_addons_5e63cb8ceb2e7e06e33b8dd7a9dcbebc0ad5ec5a_0.log)
[logs.txt](https://github.com/kubernetes/minikube/files/12068867/logs.txt)

### Operating System

Other

### Driver

Docker

**Comment by spowelljr** [comment]
Could you run `kubectl get pods -n ingress-nginx`, you should see three pods
```
NAME                                        READY   STATUS      RESTARTS   AGE
ingress-nginx-admission-create-hhx62        0/1     Completed   0          104s
ingress-nginx-admission-patch-q5tsz         0/1     Completed   1          104s
ingress-nginx-controller-7799c6795f-jxghd   1/1     Running     0          104s
```

One of them should be stuck starting, then run `kubectl describe pods <pod_name> -n ingress-nginx` & `kubectl logs <pod_name> -n ingress-nginx` and upload the output here.

**Comment by tlemons** [comment]
Hi Steven - thanks very much for the fast reply. While waiting for a response to my posting, I also tried to use k3s on the same system, which I think muddied the waters on this system, which uses SLES 12 2P5.

So, this morning, I started using a different system, running SLES 15 SP4. I followed the steps in https://github.com/ansible/awx-operator#usage and, this time, everything worked flawlessly.

So maybe my issue was caused by a specific problem on my SLES 12 SP5 system, or a general problem with that older operating system.

So I'm all set now. Thanks again for the help!
